catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/security/accesscontrol/AccessControlPreQueryPluginTest.java

```diff
@@ -0,0 +1,321 @@
+            FILTER_BUILDER.anyOf(
+                FILTER_BUILDER.attribute(Core.METACARD_TAGS).is().like().text(WORKSPACE_TAG),
+                FILTER_BUILDER.attribute(Core.METACARD_TAGS).is().like().text(QUERY_TEMPLATE_TAG)));
+    verifyPluginAddsSecurityBranchToFilter(typicalQueryForSpecificCaseOfMultipleTypes);
+  }
+
+  @Test
+  public void testDisjointQueryForSpecificCaseOfManyTypesIsOptimized()
+      throws PluginExecutionException, StopProcessingException {
+    Filter disjointQuery =
+        FILTER_BUILDER.anyOf(
+            FILTER_BUILDER.allOf(
+                FILTER_BUILDER.attribute(Core.CREATED).is().before().date(TEST_DATE),
+                FILTER_BUILDER.attribute(Core.METACARD_TAGS).is().equalTo().text(WORKSPACE_TAG)),
+            FILTER_BUILDER.allOf(
+                FILTER_BUILDER.attribute(Core.TITLE).is().like().text(TEXT_CRITERIA),
+                FILTER_BUILDER
+                    .attribute(Core.METACARD_TAGS)
+                    .is()
+                    .equalTo()
+                    .text(QUERY_TEMPLATE_TAG)),
+            FILTER_BUILDER.allOf(
+                FILTER_BUILDER.attribute(Core.ID).is().like().text(TEXT_ID),
+                FILTER_BUILDER
+                    .attribute(Core.METACARD_TAGS)
+                    .is()
+                    .equalTo()
+                    .text(ATTRIBUTE_GROUP_TAG)));
+    verifyPluginAddsSecurityBranchToFilter(disjointQuery);
+  }
+
+  @Test
+  public void testTypicalQueryForAllResourcesByUsingWildcardIsLeftAlone()
+      throws PluginExecutionException, StopProcessingException {
+    Filter typicalQueryForAllResourcesWildcard =
+        FILTER_BUILDER.attribute(Core.METACARD_TAGS).is().like().text(TEXT_WILDCARD);
+    verifyPluginDoesNotAlterTheFilter(typicalQueryForAllResourcesWildcard);
+  }
+
+  @Test
+  public void testTypicalQueryForAllResourcesByOmittingTagsAttributeIsLeftAlone()
+      throws PluginExecutionException, StopProcessingException {
+    Filter typicalQueryForAllResourcesNoTags =
+        FILTER_BUILDER.attribute(Metacard.ANY_TEXT).is().like().text(TEXT_WILDCARD);
+    verifyPluginDoesNotAlterTheFilter(typicalQueryForAllResourcesNoTags);
+  }
+
+  @Test
+  public void testTypicalQueryForSpecificCaseOfResourcesIsLeftAlone()
+      throws PluginExecutionException, StopProcessingException {
+    Filter typicalQueryForSpecificResource =
+        FILTER_BUILDER.anyOf(
+            FILTER_BUILDER.allOf(
+                FILTER_BUILDER.attribute(Core.TITLE).is().like().text(TEXT_CRITERIA),
+                FILTER_BUILDER.attribute(Core.CREATED).is().before().date(TEST_DATE)),
+            FILTER_BUILDER.allOf(
+                FILTER_BUILDER.attribute(Core.DESCRIPTION).is().like().text(TEXT_CRITERIA),
+                FILTER_BUILDER.attribute(Core.MODIFIED).is().before().date(TEST_DATE)));
+    verifyPluginDoesNotAlterTheFilter(typicalQueryForSpecificResource);
+  }
+
+  @Test
+  public void testOddQueryAcrossMultipleTypesIsLeftAlone()
```

Comment:
This one's not supposed to alter the filter because the query contains the tag 'resource' which is not in the access-controlled tags, right? If so, this test should be renamed to make that clearer because the current name doesn't mean much.